value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -2512,7 +2512,7 @@ this.[n] = [valueFromValue v][invokeSuper v].[v.names.get]()[/valueFromValue];
   [eachLine type.syntheticFieldsInjectedAnnotations]
   [jsonIgnore type]
   [if type.useLazyhash][hiddenMutableState type][/if]
-  private[if not type.serial.simple] transient[/if][if type.usePrehashed] final[/if] int hashCode;[if type.useLazyhash] // hashCode lazily computed[/if]
+  private[if ((not type.serial.simple) or type.useLazyhash)] transient[/if][if type.usePrehashed] final[/if] int hashCode;[if type.useLazyhash] // hashCode lazily computed[/if]
```

Comment:
I was thinking to remove the entire if check around `transient` but I guess we can't do that judging by the conversation in https://github.com/immutables/immutables/issues/659 @elucash?

Reply:
I was struggling to remember why `not type.serial.simple` check was put in the first place. Probably it is for precomputed hash code to be deserialized correctly when using simple(regular) serialization. So it should be correct to force transient for the `useLazyhash`. So far it seems correct solution.